file_system.go

```diff
@@ -446,6 +449,9 @@ func parseMLST(entry string, skipSelfParent bool) (os.FileInfo, error) {
 
 	if typ == "dir" || typ == "cdir" || typ == "pdir" {
 		mode |= os.ModeDir
+	} else if strings.HasPrefix(typ, "os.unix=slink") || strings.HasPrefix(typ, "os.unix=symlink") {
```

Comment:
Yes, the type may or may not be followed by a ":" which may or may not be followed by the symlink target.

I can separate the two tests and check for the colon, or check for both slink and symlink at the same time with a regex.